Rakefile

```diff
@@ -157,38 +157,71 @@ namespace :test do
     task :setup, :tf_workspace do |t, args|
       tf_workspace = args[:tf_workspace] || ENV['INSPEC_TERRAFORM_ENV']
       abort("You must either call the top-level test:azure task, or set the INSPEC_TERRAFORM_ENV variable.") unless tf_workspace
-      puts '----> Setup'
+
+      puts '----> Setup Terraform Workspace'
+
       sh("cd #{integration_dir}/build/ && terraform init -upgrade")
       sh("cd #{integration_dir}/build/ && terraform workspace new #{tf_workspace}")
 
-      # Generate Azure crendentials
-      connection = Train.create('azure').connection
-      creds = connection.options
-
-      # Determine the storage account name and the admin password
-      sa_name = (0...15).map { (65 + rand(26)).chr }.join.downcase
-      admin_password = Passgen::generate(length: 12, uppercase: true, lowercase: true, symbols: true, digits: true)
-
-      # Use the first 4 characters of the storage account to create a suffix
-      suffix = sa_name[0..3]
-
-      # Create the plan that can be applied to Azure
-      cmd  = ""
-      cmd += "cd #{integration_dir}/build/ && terraform plan -out inspec-azure.plan"
-      cmd += " -var 'subscription_id=#{creds[:subscription_id]}' "
-      cmd += " -var 'client_id=#{creds[:client_id]}' "
-      cmd += " -var 'client_secret=#{creds[:client_secret]}' "
-      cmd += " -var 'tenant_id=#{creds[:tenant_id]}' "
-      cmd += " -var 'storage_account_name=#{sa_name}' "
-      cmd += " -var 'admin_password=#{admin_password}' "
-      cmd += " -var 'suffix=#{suffix}' "
-      sh(cmd)
-
-      # Apply the plan on Azure
-      cmd = "cd #{integration_dir}/build/ && terraform apply inspec-azure.plan"
-      sh(cmd)
-
-      # Dump TF outputs to InSpec attributes file
+      Rake::Task["test:azure:vars"].execute
+      Rake::Task["test:azure:plan"].execute
+      Rake::Task["test:azure:apply"].execute
+    end
+
+    desc "Generate terraform.tfvars file"
+    task :vars do |t, args|
+
+      unless File.exist?("#{integration_dir}/build/terraform.tfvars")
```

Comment:
Maybe treat as guard statement and use `next if` instead, so the entire contents of the task doesn't have to indent under the conditional?